libs/mocks/src/pools.rs

```diff
@@ -66,12 +67,18 @@ pub mod pallet {
 			register_call!(move |(a, b, c)| f(a, b, c));
 		}
 
-		pub fn mock_benchmark_create_pool(f: impl Fn(T::PoolId, &T::AccountId) + 'static) {
+		pub fn mock_bench_create_pool(f: impl Fn(T::PoolId, &T::AccountId) + 'static) {
 			register_call!(move |(a, b)| f(a, b));
 		}
 
-		pub fn mock_benchmark_give_ausd(f: impl Fn(&T::AccountId, T::Balance) + 'static) {
-			register_call!(move |(a, b)| f(a, b));
+		pub fn mock_bench_investor_setup(
+			f: impl Fn(T::PoolId, T::AccountId, T::Balance) + 'static,
+		) {
+			register_call!(move |(a, b, c)| f(a, b, c));
+		}
+
+		pub fn mock_bench_default_investment_id(f: impl Fn(T::TrancheCurrency) + 'static) {
```

Comment:
I think the closure of this method should be `impl Fn(T::PoolId) -> T::InvestmentId + 'static` instead to match the trait method. 

This matching can only be ensured in runtime, there is no way to get the compiler warming about this. We'd need the procedural macros to avoid these problems.

Reply:
Great catch! You are right. Looking forward to having procedural macro support 🤘